src/pygeoprocessing/geoprocessing_core.pyx

```diff
@@ -941,6 +942,38 @@ def _raster_band_percentile_double(
             # you never know if this might fail!
             LOGGER.warning('unable to remove %s', file_path)
     if rm_dir_when_done:
-        shutil.rmtree(working_sort_directory)
+        _safe_rmtree(working_sort_directory)
     LOGGER.debug('here is percentile_list: %s', str(result_list))
     return result_list
+
+
+def _safe_rmtree(path):
+    """Attempts to remove a file path and logs a warning if it fails.
+
+    There is a possible bug in shutil.rmtree that can raise a FileNotFound
+    error even if the removal of a directory is possible and it removed it
+    see: https://bugs.python.org/issue29699. This function is effectively a
+    guard against that bug and logs a warning if the file path could not
+    be removed.
+
+    Args:
+        path (str): represents a file path to remove
+
+    Returns:
+        None
+    """
+    try:
+        # check if the path exists first before removing so we know if we
+        # get a FileNotFoundError we can be confident it is related to
+        # Python issue29699.
+        if os.path.exists(path):
+            shutil.rmtree(path)
+    except FileNotFoundError:
+        # check if the dir path is not there to verify if there was an
+        # incorrect exception raised due to Python issue29699. If the path
+        # does not exist assume the directory was removed correctly.
```

Comment:
Thinking about the case where I saw this from a user the problem went like this in `convolve_2d`:

1) Call for removal of `mask_dir/mask_raster_path` with `os.remove`
2) Call `shutil.rmtree` on `mask_dir`
3) `FileNotFoundError` on `mask_dir/mask_raster_path`, NOT the `mask_dir` directory path itself.

So, with this implementation we're not trying to "resolve" the python bug by then trying to "re-delete" the directory `path`, but just logging that "hey, some rare Python bug occurred so that directory `path` might still be around". And continue on our merry way.

I'm just jotting this down to make sure I'm on the same page.

Reply:
OH! I get what this is now. I thought this was some sort of internal `rmtree` race condition. I should just skip that #1 and dump this whole thing.